src/Core/Form/IdentifiableObject/DataHandler/SupplierFormDataHandler.php

```diff
@@ -0,0 +1,150 @@
+     */
+    public function __construct(
+        CommandBusInterface $commandBus,
+        ImageUploaderInterface $imageUploader
+    ) {
+        $this->bus = $commandBus;
+        $this->imageUploader = $imageUploader;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(array $data)
+    {
+        if (!isset($data['shop_association']) || !$data['shop_association']) {
+            $data['shop_association'] = [];
+        }
+
+        /** @var SupplierId $supplierId */
+        $supplierId = $this->bus->handle(new AddSupplierCommand(
+            $data['name'],
+            $data['address'],
+            $data['city'],
+            $data['id_country'],
+            $data['is_enabled'],
+            $data['description'],
+            $data['meta_title'],
+            $data['meta_description'],
+            $data['meta_keyword'],
+            $data['shop_association'],
+            $data['address2'],
+            $data['post_code'],
+            $data['id_state'],
+            $data['phone'],
+            $data['mobile_phone']
+        ));
+
+        /** @var UploadedFile $uploadedFlagImage */
+        $uploadedLogo = $data['logo'];
+
+        if ($uploadedLogo instanceof UploadedFile) {
+            $this->imageUploader->upload($supplierId->getValue(), $uploadedLogo);
+        }
+
+        return $supplierId->getValue();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update($supplierId, array $data)
+    {
+        /** @var UploadedFile $uploadedFlagImage */
+        $uploadedLogo = $data['logo'];
+        $logo = null;
+
+        if ($uploadedLogo instanceof UploadedFile) {
+            $this->imageUploader->upload($supplierId, $uploadedLogo);
+        }
+
+        $command = new EditSupplierCommand($supplierId);
+        $this->fillCommandWithData($command, $data);
+
+        $this->bus->handle($command);
+    }
+
+    /**
+     * Fills command with provided data
+     *
+     * @param EditSupplierCommand $command
+     * @param array $data
+     */
+    private function fillCommandWithData(EditSupplierCommand $command, array $data)
+    {
+        $command->setName((string) $data['name']);
+        $command->setLocalizedDescriptions($data['description']);
+        $command->setPhone($data['phone']);
+        $command->setMobilePhone($data['mobile_phone']);
+        $command->setMobilePhone($data['mobile_phone']);
+        $command->setAddress($data['address']);
+        $command->setAddress2($data['address2']);
+        $command->setPostCode($data['post_code']);
+        $command->setCity($data['city']);
+        $command->setCountryId($data['id_country']);
+        $command->setStateId($data['id_state']);
+        $command->setLocalizedMetaTitles($data['meta_title']);
+        $command->setLocalizedMetaDescriptions($data['meta_description']);
+        $command->setLocalizedMetaKeywords($data['meta_keyword']);
+        $command->setEnabled((bool) $data['is_enabled']);
```

Comment:
Same

Reply:
🆙

Reply:
Fixed castings. If we would use typecasting in methods, this would be necessary here. Now casted everything coming from form except string and arrays.